compiler/x/codegen/X86Ops.hpp

```diff
@@ -1220,6 +1219,115 @@ typedef enum
 
 class TR_X86OpCode
    {
+   enum TR_OpCodeVEX_L : uint8_t
+      {
+      VEX_L128 = 0x0,
+      VEX_L256 = 0x1,
+      VEX_L512 = 0x2,
+      VEX_L___ = 0x3, // Instruction does not support VEX encoding
+      };
+   enum TR_OpCodeVEX_v : uint8_t
+      {
+      VEX_vNONE = 0x0,
+      VEX_vReg_ = 0x1,
+      };
+   enum TR_InstructionREX_W : uint8_t
+      {
+      REX__ = 0x0,
+      REX_W = 0x1,
+      };
+   enum TR_OpCodePrefix : uint8_t
+      {
+      PREFIX___ = 0x0,
+      PREFIX_66 = 0x1,
+      PREFIX_F3 = 0x2,
+      PREFIX_F2 = 0x3,
+      };
+   enum TR_OpCodeEscape : uint8_t
+      {
+      ESCAPE_____ = 0x0,
+      ESCAPE_0F__ = 0x1,
+      ESCAPE_0F38 = 0x2,
+      ESCAPE_0F3A = 0x3,
+      };
+   enum TR_OpCodeModRM : uint8_t
+      {
+      ModRM_NONE = 0x0,
+      ModRM_RM__ = 0x1,
+      ModRM_MR__ = 0x2,
+      ModRM_EXT_ = 0x3,
+      };
+   enum TR_OpCodeImmediate : uint8_t
+      {
+      Immediate_0 = 0x0,
+      Immediate_1 = 0x1,
+      Immediate_2 = 0x2,
+      Immediate_4 = 0x3,
+      Immediate_8 = 0x4,
+      Immediate_S = 0x7,
+      };
+   template <typename TCurcor>
```

Comment:
what is the purpose of this template? I'm sure it is logical, but templates are used sparingly in TR due to the binary growth they can cause so a comment about their use in the PR would seem justified to me.